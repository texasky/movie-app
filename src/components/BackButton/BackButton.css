.back-button {
    cursor: pointer;
    width: 50px;
    background-image: url('../../assets/left-arrow.svg');
    background-position: center; 
    background-repeat: no-repeat; 
    background-size: contain;
    border-radius: 5px;
    transition: all 0.5s;
}

.back-button:hover {
    transform: scale(1.1);
    background-color: cornflowerblue;
    opacity: 0.9;
}