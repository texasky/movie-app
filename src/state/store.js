import { applyMiddleware, createStore } from "redux"
import configureStore from 'redux-mock-store'
import reducers from "./reducers/index"
import thunk from "redux-thunk";
import { composeWithDevTools } from "redux-devtools-extension";

const store = createStore(reducers, composeWithDevTools(applyMiddleware(thunk)));

export default store; 