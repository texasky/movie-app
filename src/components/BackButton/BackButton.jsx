import { useDispatch } from 'react-redux';
import { setMovieOpenToFalse } from '../../state/action-creator';
import { Link } from "react-router-dom";
import './BackButton.css'

const BackButton = (props) => {
    const dispatch = useDispatch();

    const getBackButtonHadler = () => {
        dispatch(setMovieOpenToFalse())
    }

    return (
        <Link to='/'><div className='back-button' data-testid="back-button" onClick={getBackButtonHadler}></div></Link>
    )
}

export default BackButton;