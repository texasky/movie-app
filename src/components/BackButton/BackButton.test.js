import { render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import BackButton from './BackButton';
import { Provider } from 'react-redux';
import store from './../../state/store'
import { BrowserRouter as Router } from 'react-router-dom';

test('Back button component tests', () => {
    const clickFn = jest.fn();
    const utils = render(<Provider store={store}><Router><BackButton onClick={clickFn()}/></Router></Provider>);

    const movieComponent = utils.getByTestId('back-button');
    fireEvent.click(movieComponent)

    expect(clickFn).toBeCalled();
});

test('Back button component matches snapshot', () => {
    const clickFn = jest.fn();
    const tree = renderer.create(<Provider store={store}><Router><BackButton onClick={clickFn()}/></Router></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
});