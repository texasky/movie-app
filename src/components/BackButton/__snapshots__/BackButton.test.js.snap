// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Back button component matches snapshot 1`] = `
<a
  href="/"
  onClick={[Function]}
>
  <div
    className="back-button"
    data-testid="back-button"
    onClick={[Function]}
  />
</a>
`;
